chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go

```diff
@@ -283,12 +283,30 @@ func (c *ChaosExperimentHandler) UpdateChaosExperiment(ctx context.Context, requ
 		revID = uuid.New().String()
 	)
 
-	// Check if the experiment_name exists under same project
-	err := c.validateDuplicateExperimentName(ctx, projectID, request.ExperimentName)
+	// Fetch the existing experiment
```

Comment:
This can further be optimised, by fetching the mongo document by querying it based on name. Then check if the number of documents is more than 1 then
- if request.ExperimentID == wfDetails.ExperimentID which means same document is being modified so the validation can be skipped
- if request.ExperimentID != wfDetails.ExperimentID that means another document exists with the same name so in this case it should throw an error that same name exists in the project.

 With this method you can skip using the validateDuplicateExperimentName and save an extra DB call